cmake/compiler_flags.cmake

```diff
@@ -52,7 +52,8 @@ if (CMAKE_COMPILER_IS_GNUCXX)
     -Wno-long-long 
     -Wno-unknown-pragmas
     -Wno-unused-function
-    -Wno-variadic-macros)
+    -Wno-variadic-macros
+    -msse2)
```

Comment:
this should not be required, since `For the x86-32 compiler, you must use -march=cpu-type, -msse or -msse2 switches to enable SSE extensions and make this option effective. For the x86-64 compiler, these extensions are enabled by default. `
(see https://gcc.gnu.org/onlinedocs/gcc-9.2.0/gcc/x86-Options.html).

At the moment, we only use SSE2 instructions, right?
But what if someone wants to add SSE3 code using  SIMDe? Do we need to add `-msse2 -msse3 -mssse3 -msse4 -msse4.1 -msse4.2 ` etc here manually?
Maybe add a comment what is required.